src/sjcl-ecdsa-signdeterministic.js

```diff
@@ -0,0 +1,119 @@
+'use strict';
+var sjcl = require('sjcl');
+/* eslint-disable new-cap */
+
+sjcl.ecc.ecdsa.secretKey.prototype.generateK =
+function(hash, hashObject) {
+
+  var curve = this._curve;
+  var qlen = sjcl.bitArray.bitLength(curve.r.toBits());
```

Comment:
I think `secretKey` already has the curve length cached on the property. We do care about performance somewhat :) The sjcl guys definitely do. So encoding the number as bits, then counting the length, being a little inefficient isn't ideal.